### ISSUE TITLE
rasterlayer table based on field

### ISSUE THREAD

**Issue by Derryrover** [nens/lizard-management-client#583]
Discussed on 21/1 with @Derryrover and @lexvand

- "Based on" column will be based on "is_geoblock" and contains either "Raster source" or "Geoblock" text
- [moved to #594] modal (from list view) will fist perform a request like api/v4/rastersources/?raster__uuid=80f4e80b-129a-47c2-9baa-8adfda31f042 to be able to display the name + mgmt screen link of all dependencies
- leave form as is
- "raster.raster_sources" field will disappear from API list view

**Comment by hoanphungt** [comment]
Hi @Derryrover and @caspervdw,

Do I understand correctly that this ticket is still waiting for some backend works to add the raster__uuid filter to the raster sources endpoint?

> - modal (from list view) will fist perform a request like api/v4/rastersources/?raster__uuid=80f4e80b-129a-47c2-9baa-8adfda31f042 to be able to display the name + mgmt screen link of all dependencies

**Comment by lexvand** [comment]
@hoanphungt I will create a separate issue for the modal. First perform the other more basic changes.

**Comment by Derryrover** [comment]
> Hi @Derryrover and @caspervdw,
> 
> Do I understand correctly that this ticket is still waiting for some backend works to add the raster__uuid filter to the raster sources endpoint?
> 
> > * modal (from list view) will fist perform a request like api/v4/rastersources/?raster__uuid=80f4e80b-129a-47c2-9baa-8adfda31f042 to be able to display the name + mgmt screen link of all dependencies

@hoanphungt 
Yes this is true. You could try to proceed by using the url "api/v4/rastersources/" instead to get started.

**Comment by hoanphungt** [comment]
@Derryrover 
Hi Tom,

Could you review this PR: https://github.com/nens/lizard-management-client/pull/597? This is to address bullet point 1. We already have a separate ticket for the modal: https://github.com/nens/lizard-management-client/issues/594. Bullet point 3 and 4 require no action right now. However, it is noted that if the "raster.raster_sources" field is to be removed from the API list view of rasters then this will break the raster layer form. So preferably, we need to take actions on the raster layer form as well.

**Comment by Derryrover** [comment]
@hoanphungt 

 https://github.com/nens/lizard-management-client/pull/597 
looks good

I am proceeding with:
https://github.com/nens/lizard-management-client/issues/594

**Comment by Derryrover** [comment]
@hoanphungt 
@lexvand 

I see that https://github.com/nens/lizard-management-client/issues/594 is not picked up yet.
I am very hesitant to this change the based_on column if it is not in combination with a modal.
why?
Because it is not an improvement on the current situation.


**Alternative**
I think an alternative to show the modal could be that:

- When we open the modal it retrieves the detail view of the raster layer
- on this detail view api response all related raster-sources urls are shown.
- we use the urls, to exclude uuids.
- this data is used in the modal. We could show the uuid, but make it a hyperlink to the relevant management page.
Here was a previous draft I made:
https://github.com/nens/lizard-management-client/pull/582/files

**Comment by hoanphungt** [comment]
@Derryrover 
Hi Tom,

I don't agree to implement the modal together with this change. The reasons are:

- The "raster_sources" field will be deleted from the API list view so the modal you suggest to make will soon be broken.
- The current situation of the raster layer table will also be broken once the "raster_sources" field is removed from the API.
- The new modal https://github.com/nens/lizard-management-client/issues/594 is currently being blocked by backend work and is not ready to work on yet. We can wait for it to be on staging before start working on it or we can also already start working on it in parallel (although I rather not).

So by splitting them into 2 separate tasks, we can improve the current situation slightly (the URL now is not useful at all) while avoid doing double works for the modal later (build the modal now and later on fix it to adapt to new changes from the API).

**Comment by lexvand** [comment]
@hoanphungt I think we should remove ordering on the Based on column. It is not supported by the backend and I do not see this as something useful.

**Comment by hoanphungt** [comment]
Removed ordering in "Based On" column of raster layer table

**Comment by GeOdin** [comment]
@hoanphungt 

Hey Hoan,

I saw that the 'Based on' column of the raster layer table now contains 'Raster source' or Geoblock.
The 'Based on' column can no longer be ordered on.
The uuid of the raster source/ geoblock is nicely shown as data source in the raster layer form and when clicking on 'Raster source' or Geoblock in the 'Basted on' column of the raster layer table.

Do I then cover everything of this ticket or am I still missing something?

Kind regards,
Madeleine

**Comment by hoanphungt** [comment]
@GeOdin 
Yes, you are correct!

**Comment by GeOdin** [comment]
@hoanphungt 

Okay, thank you. :)
